### PR TITLE
Add long file paths solution to Windows troubleshooting

### PR DESCRIPTION
**Pull request by MolotovCherry**
Currently, building on Windows may fail for people who have not enabled long filepaths on Windows itself.

This PR adds the solution to the troubleshooting section in the `Building Zed for Windows` guide.

For an example failure error message:
```rs
error: failed to get `pet` as a dependency of package `languages v0.1.0 (D:\a\zed-windows-builds\zed-windows-builds\crates\languages)`

Caused by:
  failed to load source for dependency `pet`

Caused by:
  Unable to update https://github.com/microsoft/python-environment-tools.git?rev=ffcbf3f28c46633abd5448a52b1f396c322e0d6c#ffcbf3f2

Caused by:
  path too long: 'C:/Users/runneradmin/.cargo/git/checkouts/python-environment-tools-903993894b37a7d2/ffcbf3f/crates/pet-conda/tests/unix/conda_env_without_manager_but_found_in_history/some_other_location/conda_install/conda-meta/python-fastjsonschema-2.16.2-py310hca03da5_0.json'; class=Filesystem (30)
```

Release Notes:

- N/A

### CODE REVIEW

**Comment by MolotovCherry** [comment]
I ran `pnpm dlx prettier windows.md --write` on the file locally, but no changes were made which I can upload according to git. Any clue?

File: docs/src/development/windows.md [hunk 1]
```diff
@@ -120,3 +120,32 @@ In order to fix this issue, you can manually set the `ZED_RC_TOOLKIT_PATH` envir
 `C:\Program Files (x86)\Windows Kits\10\bin\<SDK_version>\x64`.
 
 See this [issue](https://github.com/zed-industries/zed/issues/18393) for more information.
+
+### Build fails: Path too long
+
+You may receive an error like the following when building
+
+```rs
```
Comment (SomeoneToIgnore) [inline docs/src/development/windows.md:128]:
```suggestion
```
```

Comment (SomeoneToIgnore) [reply docs/src/development/windows.md:128]: It's not a Rust code block, so let's remove it.

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you.

**Comment by ReturnMars** [comment]
thanks

**Pull request merged**